packages/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 set(PACKAGE "libyang")
 set(CPP_PACKAGE "libyang-cpp")
 set(PYTHON_PACKAGE "python3-yang")
```

Comment:
Please, modify these variables instead of adding `1` for all their uses, that is what they are here for.

Reply:
I've considered this and tried it initially. But the challenge is that the rename should not happen for all packages.
Currently the packages (before the change) are:
* libyang-1.X.YYY-0.src.rpm
* libyang-1.X.YYY-0.x86_64.rpm
* libyang-cpp-1.X.YYY-0.x86_64.rpm
* libyang-cpp-debuginfo-1.X.YYY-0.x86_64.rpm
* libyang-cpp-devel-1.X.YYY-0.x86_64.rpm
* libyang-debuginfo-1.X.YYY-0.x86_64.rpm
* libyang-debugsource-1.X.YYY-0.x86_64.rpm
* libyang-devel-1.X.YYY-0.x86_64.rpm
* python3-yang-1.X.YYY-0.x86_64.rpm
* python3-yang-debuginfo-1.X.YYY-0.x86_64.rpm

In Debian (and similar to RedHat), I do believe that the devel packages should not be renamed, so the correct new naming should be:
* libyang1-1.X.YYY-0.src.rpm
* libyang1-1.X.YYY-0.x86_64.rpm
* libyang-cpp1-1.X.YYY-0.x86_64.rpm
* libyang-cpp1-debuginfo-1.X.YYY-0.x86_64.rpm
* libyang-cpp-devel-1.X.YYY-0.x86_64.rpm
* libyang1-debuginfo-1.X.YYY-0.x86_64.rpm
* libyang1-debugsource-1.X.YYY-0.x86_64.rpm
* libyang-devel-1.X.YYY-0.x86_64.rpm
* python3-yang1-1.X.YYY-0.x86_64.rpm
* python3-yang1-debuginfo-1.X.YYY-0.x86_64.rpm
* yang-tools-1.X.YYY-0.x86_64.rpm
* yang-tools-debuginfo-1.X.YYY-0.x86_64.rpm

So basically, the libyang-devel (and the other devel) packages as well as the `yang-tools` should not be renamed. This supports the install of the binary (runtime) of both versions, but only latest on development.

If I change the `packages/CMakeLists.txt`, then the dependencies will get a bit messy, eg in
```
%package -n @CPP_PACKAGE@-devel
Summary:    Headers of bindings to c++ language
Requires:   @CPP_PACKAGE@ = %{version}-%{release}
Requires:   pcre-devel
```
Here I would need to use a 2nd `CPP_PACKAGE_WITHVERSION` or something similar name to reflect that `libyang-cpp-devel` depends on `libyang-cpp1` (and not on `libyang-cpp`). Similar on the other devel packages. Could be done, but looked messy to me.

Additionally, if someone uses the CMakeLists file to rename the packages, this would cause a much greater complication to get it right.

So based on this, I decided to go for idea to add version number directly in the rpm spec file.

But happy to change this over if this is still you preference.

Reply:
Oh, I see. Well, there is a reason why I hate all the packaging stuff. Anyway, I would then ask you to just replace `1` with `@LIBYANG_MAJOR_SOVERSION@` so that we can reuse these scripts for the new *libyang*, unless you think there is some other problem, thanks.

Reply:
Good suggestion. Not sure why I missed that variable before.
Done in latest update